bin/_os.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+set -eu
+
+os() {
+  os=$(uname -s)
+  arch=""
```

Comment:
Do we need to set this if it's only used in the `Linux` case and gets set right away by `arch=$(unman -m)`?

Reply:
Yes this one is necessary. Because if we remove it, when it is not Linux, then it will throw error `arch: unbound variable` on `if [ -n "$arch" ]; then`

Reply:
Oh okay yea I wasn't sure if `-n` would evaluate to false when it wasn't set _or_ unbound. Thanks!